merkle/src/merkle.rs

```diff
@@ -248,6 +258,237 @@ impl<E: Element> Clone for MmapStore<E> {
     }
 }
 
+/// File-mapping version of `MmapStore` with the added `new_with_path` method
+/// that allows to set its path (otherwise a temporary file is used which is
+/// cleaned up after we drop this structure).
+#[derive(Debug)]
+pub struct DiskMmapStore<E: Element> {
+    // Implementing the `store` with `Arc`/`RwLock` to avoid adding lifetimes
+    // parameters to the struct (which might have larger repercussions in the
+    // definitions of the `MerkleTree` and its consumers. Also used for its
+    // coordination mechanisms, but it's not clearly defined if `MerkleTree`
+    // (and `Store`) should be thread-safe (so they might be removed later).
+    store: Arc<RwLock<Option<MmapMut>>>,
+
+    len: usize,
+    _e: PhantomData<E>,
+    file: File,
+    // We need to save the `File` in case we're creating a `tempfile()`
+    // otherwise it will get cleaned after we return from `new()`.
+
+    // We cache the `store.len()` call to avoid accessing the store when
+    // it's offloaded. Not to be confused with `len`, this saves the total
+    // size of the `store` and the other one keeps track of used `E` slots
+    // in the `DiskMmapStore`.
+    store_size: usize,
+
+    // We save the arguments of `new_with_path` to reconstruct it and reload
+    // the `MmapMut` after offload has been called.
+    path: String,
+    // FIXME: Using `String` instead of `&Path` to avoid using lifetimes
+    //  or errors like "the size for values of type `[u8]` cannot be known
+    //  at compilation time" (with `Option<Box<Path>>` or similar).
+    size: Option<usize>,
+    // FIXME: Needed to reconstruct the `new_with_path` call, especially
```

Comment:
does this still need fixing?

Reply:
If possible I'd want to get the size from the MT itself, if the `Store` could access it, if not, then remembering `size` here is fine, WDYT?

Reply:
Could we pass the size from the tree, down to the relevant methods when calling them?

Reply:
`new_with_path` is (also) used internally when regenerating the `mmap` after if has been offloaded so there's no source to pass down the size from, so on a second thought keeping this here seems fine, marking as fixed.